src/NeoSharp.Core/Network/Server.cs

```diff
@@ -131,9 +131,9 @@ public void ConnectToPeers(params EndPoint[] endPoints)
         #region IBroadcaster implementation
 
         /// <inheritdoc />
-        public void Broadcast(Message message, IPeer source)
+        public async Task Broadcast(Message message, IPeer source)
         {
-            Parallel.ForEach(_connectedPeers.Where(p => p != source), p => p.Send(message));
+            await Task.Factory.StartNew(() => Parallel.ForEach(_connectedPeers.Where(p => p != source), p=> p.Send(message)));
```

Comment:
Once again, it is the same as Task.CompleteTask and we do not need to await on broadcasting.

Reply:
So what is the right way to do this in async way?

Reply:
Send is async method. It does not block the thread. Parallel.ForEach makes sure Send is called for all peers at the same time. My idea was we don't want to wait until broadcasting will be completed. That is why we call this (async) and do not wait until all these operations will be done. The void result prevents from awaiting and blocking the current thread.